utilities/cli/version_check.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Version checking utility for holohub CLI.
+"""
+
+import os
+import subprocess
+import time
+from pathlib import Path
+
+
+def check_for_cli_updates() -> None:
+    """
+    Check if a newer version of the CLI is available.
+
+    - Checks once per 24 hours (configurable via CLI_CHECK_INTERVAL)
```

Comment:
That seems a bit overengineered? I was hoping of something simpler:
- if using a git tag or specific git ref, ensure the sha requested matches the sha cached
- if using a branch, only check if there is a more recent sha than the sha cached if explicitly asking to update